### PR TITLE
Fix cannotBeEmpty bug and add some tests

### LINKED ISSUE #640: is not applicable to concrete nodes at path "sonata_customer.profile.menu."
Composer update failed with error 
```

ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!
!!  In ArrayNodeDefinition.php line 475:
!!                                                                              
!!    ->cannotBeEmpty() is not applicable to concrete nodes at path "sonata_customer.profile.menu."
!!                                                                              
!!
!!
Script @auto-scripts was called via post-update-cmd

```

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because it is patch.

Closes #640 

## Changelog

```markdown
### Fixed
->cannotBeEmpty() is not applicable to concrete nodes at path "sonata_customer.profile.menu."
```

### CODE REVIEW

File: tests/CustomerBundle/DependencyInjection/ConfigurationTest.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CustomerBundle\Tests\DependencyInjection;
+
+use Matthias\SymfonyConfigTest\PhpUnit\ConfigurationTestCaseTrait;
+use PHPUnit\Framework\TestCase;
+use Sonata\CustomerBundle\DependencyInjection\Configuration;
+
+class ConfigurationTest extends TestCase
```
Comment (core23) [inline tests/CustomerBundle/DependencyInjection/ConfigurationTest.php]:
```suggestion
final class ConfigurationTest extends TestCase
```

File: tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        $this->load([
+            'profile' => [
+                'menu' => null,
+                'blocks' => null,
+            ],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
```
Comment (core23) [inline tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php]: We don't use this anymore

File: tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+     */
+    protected function getMinimalConfiguration()
```
Comment (core23) [inline tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php]: Can you add a return type here?

File: src/CustomerBundle/Block/ProfileMenuBlockService.php [hunk 4]
```diff
@@ -54,7 +54,7 @@ public function configureSettings(OptionsResolver $resolver): void
 
         $resolver->setDefaults([
             'cache_policy' => 'private',
-            'menu_template' => 'SonataBlockBundle:Block:block_side_menu_template.html.twig',
+            'menu_template' => '@SonataBlockBundle/Block/block_side_menu_template.html.twig',
```
Comment (phansys) [inline src/CustomerBundle/Block/ProfileMenuBlockService.php]:
```suggestion
            'menu_template' => '@SonataBlock/Block/block_side_menu_template.html.twig',
```

File: tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setParameter('kernel.bundles', ['SonataCustomerBundle' => true]);
+    }
+
+    /**
+     * @group legacy
```
Comment (phansys) [inline tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php]: Are the methods marked with "legacy" group intended to be kept in the next major release?

File: tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    protected function load(array $configurationValues = []): void
+    {
+        $configs = [$this->getMinimalConfiguration(), $configurationValues];
+
+        foreach ($this->container->getExtensions() as $extension) {
+            if ($extension instanceof PrependExtensionInterface) {
+                $extension->prepend($this->container);
+            }
+
+            $extension->load($configs, $this->container);
```
Comment (phansys) [inline tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php]: Is required to call `load()` on every iteration?

Comment (wbloszyk) [reply tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php]: Yes, some test have diffrence configs. Check methods: `testTwigConfigParameterIsNotSet` and `testEmptyProfile`.

File: tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php (hunk 5 above)
Comment (wbloszyk) [reply tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php]: Methods marked with "legacy" should be rewrite in next major. They are from older version (based on SonataUserBundle 3.x).

Comment (phansys) [reply tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php]: Then, please add the corresponding `NEXT_MAJOR` comments.

Comment (wbloszyk) [reply tests/CustomerBundle/DependencyInjection/SonataCustomerExtensionTest.php]: I update test. Now `NEXT_MAJOR` is not required.

File: src/CustomerBundle/Block/RecentCustomersBlockService.php [hunk 7]
```diff
@@ -61,12 +64,12 @@ public function execute(BlockContextInterface $blockContext, Response $response
         ], $response);
     }
 
-    public function validateBlock(ErrorElement $errorElement, BlockInterface $block): void
+    public function validateBlock(ErrorElement $errorElement, BlockInterface $block)
```
Comment (greg0ire) [inline src/CustomerBundle/Block/RecentCustomersBlockService.php]: Why are you removing all these type declarations? :thinking:

Comment (wbloszyk) [reply src/CustomerBundle/Block/RecentCustomersBlockService.php]: I dont want change parent declarations type when someone can extends this class. If class will be final then I will add declaration. (Final will be add in next major).

Comment (greg0ire) [reply src/CustomerBundle/Block/RecentCustomersBlockService.php]: But… you _are_ changing the declaration, so the extending classes have no choice but to have `: void` right now, don't they? The `: void` has been here for more than 2 years, it was introduced in 836b7eae7eaaec832cb5c7834723f1fb040b5a4f

Comment (wbloszyk) [reply src/CustomerBundle/Block/RecentCustomersBlockService.php]: Look at: https://3v4l.org/bTGCg
I know it is not good change for code, but it eliminate posibillity of Fatal Error when extends this block

Comment (greg0ire) [reply src/CustomerBundle/Block/RecentCustomersBlockService.php]: >  but you cant remove it when parent have it.

Yeah so right now, every class that exists that extends this one has `: void` here. Please revert this change, it's only a loss of type information, and encourages people to not provide it when extending, which can only be worse (less information for SA tools)

Comment (wbloszyk) [reply src/CustomerBundle/Block/RecentCustomersBlockService.php]: Ok. I add `NEXT_MAJOR: make this class final`.

**Pull request merged**